### PR TITLE
XSpeeds

### PR DESCRIPTION
**Pull request by twistedroutes**
Updated the pull request with a new change because of the query not handling the cloudflare DDoS either.

### CODE REVIEW

**Comment by JigSawFr** [comment]
:+1: nice work

**Comment by twistedroutes** [comment]
Sorry to add to the pull again, but I grabbed the wrong DL link originally for bitsoup, so I corrected that.

File: src/Jackett/Indexers/BitSoup.cs [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+                // Tracker cannot search multi categories
+                // so we either search "all"
+                // or do multiple searches
+                if (trackerCats.Count == 0)
+                {
+                    searchUrl += "?" + queryCollection.GetQueryString();
+                    await ProcessPage(releases, searchUrl);
+                } else
+                {
+                    foreach (var cat in trackerCats)
```
Comment (Azerelat) [inline src/Jackett/Indexers/BitSoup.cs]: A little concerned that if you search for TV or a Movie generically its going to hit the server 5 times.  If you are searching for something I would go for everything as the odds are you will get the item you are searching for then any extraneous items not in the correct category will be removed by the parent controller which calls this.

**Comment by twistedroutes** [comment]
Agree.  That is indeed what I wrote in the commit:
 - 	For multi-category searches will perform multiple searches as the tracker search didn't support multiple categories. Recommend to the user to search for "all" instead. Consider changing this approach later.

So we should just treat everything incoming (for this indexer, with multiple categories in the search) to be category 0, yes?

**Comment by Azerelat** [comment]
Yep, its also showing a conflict preventing the merge.

**Comment by twistedroutes** [comment]
Does it show you the conflict? .  All I see is "this branch has conflicts that must be resolved.  Only those with write access to this repository can merge pull requests"

**Comment by twistedroutes** [comment]
Fixed the conflict, it was your edits to Startup.cs so I brought yours into my fork, remerged and then updated Startup.cs again to include the now missing ProxyConnection.

I also improved the category search for bitsoup as discussed as well as improving the accumulate cookies roundtrips (mostly for for xspeeds).

**Pull request merged**